src/forms/Button.jsx

```diff
@@ -13,29 +24,69 @@ class Button extends React.PureComponent {
 
 	render() {
 		const {
-			// removing reset & fullWidth props from other to prevent an invalid boolean being passed to <button>
-			fullWidth, // eslint-disable-line no-unused-vars
+			fullWidth,
 			icon,
 			hasHoverShadow, // eslint-disable-line no-unused-vars
 			component,
+			to,
 			...other
 		} = this.props;
 
 		// checking for the icon component signature if passed as icon={<Icon shape="search"/>} and grabbing the shape prop
-		// since this functionality is built into button in order to enforce strict sizes on button icons
-		const props = { icon: icon && icon.props && icon.props.shape };
+		const iconShape = icon && icon.props && icon.props.shape;
+
+		// this.props.component is either a string or a Link component with a `component.name` prop
+		const componentName = component
+			? component.name
+				? component.name
+				: component
+			: undefined;
 
-		if (component !== undefined && component !== 'button' && component !== 'a') {
-			console.warn(
-				'All Swarm UI v2 Button components are button elements. Future iterations will support links as independent components and the logic will be handled here'
+		if (componentName && !BUTTON_COMPONENT_TYPES.includes(componentName)) {
+			console.error(
+				'Invalid component prop for <Button>. All Swarm UI v2 Button components are button, anchor, or Link elements.'
 			);
 		}
 
-		if (component === 'a') {
-			return <SwarmLink {...props} grow={fullWidth} {...other} />;
+		// support for react-router Link component
+		if (componentName === 'Link' && to !== undefined) {
+			const { children, right, ...linkProps } = other;
+			const buttonType = getButtonType(this.props);
+			const color = FILLS[buttonType];
+
+			return (
+				<Link
+					data-swarm-button={buttonType}
+					data-swarm-size={getSwarmSize(this.props)}
+					data-icon={getIconPosition(this.props)}
+					data-swarm-width={fullWidth ? 'grow' : 'default'}
+					to={to}
+					{...linkProps}
+				>
+					{icon ? (
+						<span>
+							{right ? (
+								<React.Fragment>
+									{children}
+									<Icon shape={iconShape} size="xs" color={color} />
+								</React.Fragment>
+							) : (
+								<React.Fragment>
+									<Icon shape={iconShape} size="xs" color={color} />
+									{children}
+								</React.Fragment>
+							)}
+						</span>
```

Comment:
consider moving this deeply-nested ternary into a pure function component that can be called like

```jsx
icon
  ? <ButtonIcon right={right} shape={iconShape} color={color}>{children}</ButtonIcon>
  : children
```

Alternatively, there are some cool ways of [using flexbox to reverse the layout order of elements](https://developer.mozilla.org/en-US/docs/Web/CSS/flex-direction) without changing markup, and that would be a pretty slick solution that would making the layout rule affect the JSX

Reply:
This is pretty much 100% copy/paste code from swarm-ui:

https://github.com/meetup/swarm-ui/blob/master/packages/swarm-components/src/Button.jsx#L107

But! I like this idea as an improvement for swarm-ui. 

As mentioned in this previous comment, this code should hopefully only be temporary and get replace by a `<SwarmButton component={Link}>` or something similar in a future iteration.